x-pack/plugins/security_solution/docs/testing/test_plans/detection_response/prebuilt_rules/upgrade.md

```diff
@@ -500,242 +233,167 @@ And user should see a CTA to upgrade prebuilt rules
-And there is at least 1 rule available to install
-And this rule has neither Setup guide nor Investigation guide
-When user opens the Add Rules page
-Then all rules available for installation should be displayed in the table
-When user opens the rule preview for this rule
-Then the preview should open
-And the Setup Guide section should NOT be displayed in the Overview tab
-And the Investigation Guide tab should NOT be displayed
-```
-
-### Rule installation workflow: filtering, sorting, pagination
-
-TODO: add scenarios https://github.com/elastic/kibana/issues/166215
-
-### Rule installation workflow: misc cases
-
-#### **Scenario: User opening the Add Rules page sees a loading skeleton until the package installation is completed**
-
-**Automation**: unit tests.
+Given X prebuilt rules are installed in Kibana
+And for Y of the installed rules there are new versions available
+And all of those Y new versions have conflicts with the current versions
+And user is on the Rule Management page
+When user opens the Rule Updates table
+Then Y rules available for upgrade should be displayed in the table
+When user selects <Z> rules, all of which have upgrade conflicts
+Then user should see a CTA to upgrade <Z> number of rules, which should be disabled
+When user hovers on the CTA to upgrade multiple rules
+Then a message should be displayed that informs the user why the rules cannot be updated
 
-```Gherkin
-Given prebuilt rules package is not installed
-When user opens the Add Rules page
-Then user should see a loading skeleton until the package installation is completed
+Examples:
+  | Z                               |
+  | a few rules on the page, e.g. 2 |
+  | all rules on the page, e.g. 12  |
 ```
 
-#### **Scenario: User can navigate from the Add Rules page to the Rule Management page via breadcrumbs**
+#### **Scenario: User can upgrade all available conflict-free prebuilt rules at once**
 
-**Automation**: 1 e2e test.
+**Automation**: 1 e2e test with mock rules + integration tests with mock rules that would test /status and /upgrade/\* endpoints in integration.
 
 ```Gherkin
-Given user is on the Add Rules page
-When user navigates to the Rule Management page via breadcrumbs
-Then the Rule Management page should be displayed
+Given X prebuilt rules are installed in Kibana
+And for Y of the installed rules there are new versions available
+And those Y new versions don't have conflicts with the current versions
+And user is on the Rule Management page
+When user opens the Rule Updates table
+Then Y rules available for upgrade should be displayed in the table
+When user upgrades all rules
+Then success message should be displayed after upgrade
+And user should NOT see a CTA to upgrade prebuilt rules
+And user should NOT see a number of rules available to upgrade
 ```
 
-### Rule upgrade workflow: base cases
+#### **Scenario: User cannot upgrade all prebuilt rules at once if they have upgrade conflicts**
```

Comment:
```suggestion
#### **Scenario: User cannot upgrade all prebuilt rules at once if some rules upgrade conflicts**
```

Reply:
No, this scenario is when ALL rules have conflicts, and the Update All button is disabled